positron/modules/atom_common_id_weak_map.js

```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+const { classes: Cc, interfaces: Ci, results: Cr, utils: Cu } = Components;
+
+// Described In Electron's id_weak_map.h as:
+// "Like ES6's WeakMap, but the key is Integer and the value is Weak Pointer."
+
+function IDWeakMap() {
+  const map = new Map();
+  this.set = function(key, value) {
+    map.set(key, Cu.getWeakReference(value));
+  };
+  this.get = function(key) {
+    return map.get(key).get();
+  }
+  this.has = map.has.bind(map);
+  this.clear = map.clear.bind(map);
+}
```

Comment:
> It might. We should check if anyone has used the Electron version by calling the methods by going through the prototype.

It's intended to be an internal API, so I wouldn't expect much usage outside Electron, which doesn't itself call these methods via the prototype. This GitHub search finds some Electron forks (including ours), plus a couple of Node forks (that presumably use a different implementation—ours is from Electron, not Node), but none of those use the prototype either:

https://github.com/search?q=IDWeakMap&type=Code&utf8=%E2%9C%93

Nevertheless, it's trivial to change to the implementation that preserves the prototype chain a caller would expect, so I'll make the change!